plugins/discourse-calendar/assets/javascripts/discourse/pre-initializers/rich-editor-extension.js

```diff
@@ -17,8 +16,105 @@ const EVENT_ATTRIBUTES = {
   recurrenceUntil: { default: null },
   chatEnabled: { default: null },
   chatChannelId: { default: null },
+  customFields: { default: null },
 };
 
+const KNOWN_ATTRIBUTES = new Set(Object.keys(EVENT_ATTRIBUTES));
+let siteSettings;
+let cachedCustomFieldSetting;
+let cachedCustomFieldConfig;
+
+function getCustomFieldConfig() {
+  const setting =
+    siteSettings?.discourse_post_event_allowed_custom_fields ?? "";
+
+  if (cachedCustomFieldConfig && cachedCustomFieldSetting === setting) {
+    return cachedCustomFieldConfig;
+  }
+
+  const datasetToOriginal = new Map();
+  const originalToDataset = new Map();
+
+  setting
+    .split("|")
+    .map((field) => field.trim())
+    .filter(Boolean)
+    .forEach((field) => {
+      const datasetKey = toDatasetKey(field);
+      if (!datasetToOriginal.has(datasetKey)) {
+        datasetToOriginal.set(datasetKey, field);
+      }
+      if (!originalToDataset.has(field)) {
+        originalToDataset.set(field, datasetKey);
+      }
+    });
+
+  cachedCustomFieldSetting = setting;
+  cachedCustomFieldConfig = {
+    datasetToOriginal,
+    originalToDataset,
+  };
+
+  return cachedCustomFieldConfig;
+}
+
+function toDatasetKey(fieldName) {
+  if (!fieldName) {
+    return fieldName;
+  }
+
+  const normalized = fieldName
+    .replace(/[-_]+([a-zA-Z0-9])/g, (_, char) => char.toUpperCase())
```

Comment:
I am not sure if this is exactly camelize or not, Maybe we swap back with camelize? I worry we are missing some edge case around -_ handling without this.